src/compileFunction.js

```diff
@@ -0,0 +1,100 @@
+import parseValue from './parseValue'
+import extractFunctionSpecFromDoc from './extractFunctionSpecFromDoc'
+
+// TODO: bring back features from the original implementation.
+export default function compileFunction (name, decl, code, commentBlocks) {
+  let params = []
+  for (let node of decl.params) {
+    let param = {}
+    switch (node.type) {
+      case 'Identifier':
+        // TODO: what is an 'extensible' parameter?
```

Comment:
Yes, I made that one up :) It's means repeatable named arguments. So a function with an `extensible` parameter is allowed to be called with an arbitrary number of arbitrarily named arguments.  A trivial example would be an object factory function like `object(x=1, y=4)`. A more useful example be something like `extend(table, log_x = log(.x))` which would add a new column to the table - like R's `mutate` function https://dplyr.tidyverse.org/reference/mutate.html.